src/Algebra/Structures.agda

```diff
@@ -542,14 +542,33 @@ record IsCancellativeCommutativeSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a 
 
   open IsCommutativeSemiring isCommutativeSemiring public
 
-
-record IsKleeneAlgebra (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ) where
+record IsIdempotentSemiring (+ * : Op₂ A) (0# 1# : A) : Set (a ⊔ ℓ) where
   field
-    isSemiring  : IsSemiring + * 0# 1#
-    +-idem      : Idempotent +
+    isSemiring     : IsSemiring + * 0# 1#
+    +-idem         : Idempotent +
 
   open IsSemiring isSemiring public
 
+record IsKleeneAlgebra (+ * : Op₂ A) (⋆ : Op₁ A) (0# 1# : A) : Set (a ⊔ ℓ) where
+  field
+    isIdempotentSemiring  : IsIdempotentSemiring + * 0# 1#
+    starExpansive         : StarExpansive 1# + * ⋆
+    starDestructive       : StarDestructive + * ⋆
+
+  open IsIdempotentSemiring isIdempotentSemiring public
+
+  starLeftExpansive : StarLeftExpansive 1# + * ⋆
```

Comment:
The standard way of naming these is `starExpansiveˡ` and `starExpansiveʳ`?

Reply:
Done :)